sycl/include/sycl/accessor.hpp

```diff
@@ -1209,7 +1209,12 @@ class __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   using reference = DataT &;
   using const_reference = const DataT &;
 
-  using iterator = typename detail::accessor_iterator<DataT, Dimensions>;
+  // When accessor is read-only, we should not be able to modify it through the
+  // iterator and therefore, effectively returning const_iterator here
+  using iterator = typename detail::accessor_iterator<
```

Comment:
Just a question, maybe we should define the `value_type` typename this way as described in https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_interface_for_buffer_host_accessors?

Reply:
Good catch, fixed in 36bb61c7bedaa74df0f57d977140aa37022dc9b5